### PR TITLE
feat: refactor the remaining components to composition API

### LINKED ISSUE #124: Implement Composition API in app components
Implement Composition API in app components.

### PR DESCRIPTION
**Pull request by lukeromanowicz**
## Description
Resolves #124 

Rewrite all remaining components to Composition API

## Demo
No visual changes
image

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] My change does not require a change to the documentation.

### CODE REVIEW

**Comment by janmichek** [comment]
I invited @csiarab to test here. Here was about rewriting components to a modern approach. Please go through the components listed in Files Changed. There should be no change in functionality. Thanks for the collaboration .)

File: src/components/MicroblocksSequence.vue [hunk 1]
```diff
@@ -7,7 +7,7 @@
       tag="div">
       <div
         v-for="microblock in microblocks"
-        :key="microblock.hash"
+        :key="microblock.hash + microblock.prev_key_hash"
```
Comment (janmichek) [inline src/components/MicroblocksSequence.vue:10]: What is this needed for? I would expect some explanation here as it is out of the scope of initial issue.

File: src/components/TheHeader.vue [hunk 2]
```diff
@@ -58,38 +58,33 @@
...
+})
+
+onBeforeUnmount(() => {
+  if (isDesktop()) {
+    window.removeEventListener('resize', closeNavigation())
+  }
+})
+
+function toggleNavigation() {
+  isNavigationOpen.value = !(isNavigationOpen.value)
```
Comment (janmichek) [inline src/components/TheHeader.vue:84]: are the brackets needed in here?

**Review by janmichek — APPROVED** [review]
Cool, thanks for the detailed work. I like the change .) Less code is better.
I have not spotted any issues when navigating over the app.
I just have a few questions, but glad to approve this 🦖

File: src/components/MicroblocksSequence.vue (hunk 1 above)
Comment (lukeromanowicz) [reply src/components/MicroblocksSequence.vue:10]: You are right, this change deserves some extra explanation:

Unrelated fix for microblocks with the same hash that are created twice with the same hash due to microfork happening on new keyblock mint

**Review by michele-franchi — APPROVED** [review]
The code looks really good 👍 Let's do a quick test

**Review by csiarab — APPROVED** [review]

**Pull request merged**